File: scaffold/src/main/java/com/vgaw/scaffold/view/TxtTxtLayout.java
package com.vgaw.scaffold.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.vgaw.scaffold.R;
import com.vgaw.scaffold.util.phone.DensityUtil;

public class TxtTxtLayout extends LinearLayout {
    private TextView mTxtTxtUp;
    private TextView mTxtTxtBelow;

    public TxtTxtLayout(Context context) {
        super(context);
        init(null);
    }

    public TxtTxtLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(attrs);
    }

    public TxtTxtLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(attrs);
    }

    public void setUpText(String up) {
        mTxtTxtUp.setText(up);
    }

    private void init(AttributeSet attrs) {
        setOrientation(VERTICAL);
        setGravity(Gravity.CENTER);

        View view = View.inflate(getContext(), R.layout.txt_txt_layout, this);
        mTxtTxtUp = view.findViewById(R.id.txt_txt_up);
        mTxtTxtBelow = view.findViewById(R.id.txt_txt_below);

        if (attrs != null) {
            TypedArray array = getContext().obtainStyledAttributes(attrs, R.styleable.TxtTxtLayout);
            String txtUp = array.getString(R.styleable.TxtTxtLayout_txtUp);
            String txtBelow = array.getString(R.styleable.TxtTxtLayout_txtBelow);
            int txtUpSize = array.getDimensionPixelSize(R.styleable.TxtTxtLayout_txtUpSize, 0);
            int txtBelowSize = array.getDimensionPixelSize(R.styleable.TxtTxtLayout_txtBelowSize, 0);
            int txtUpColor = array.getColor(R.styleable.TxtTxtLayout_txtUpColor, Color.BLACK);
            int txtBelowColor = array.getColor(R.styleable.TxtTxtLayout_txtBelowColor, Color.BLACK);
            int padding = array.getDimensionPixelSize(R.styleable.TxtTxtLayout_txtPadding, 0);

            array.recycle();

            mTxtTxtUp.setText(txtUp);
            mTxtTxtUp.setTextSize(DensityUtil.px2sp(getContext(), txtUpSize));
            mTxtTxtUp.setTextColor(txtUpColor);

            mTxtTxtBelow.setText(txtBelow);
            mTxtTxtBelow.setTextSize(DensityUtil.px2sp(getContext(), txtBelowSize));
            mTxtTxtBelow.setTextColor(txtBelowColor);
            LayoutParams layoutParams = new LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
            layoutParams.topMargin = padding;
            mTxtTxtBelow.setLayoutParams(layoutParams);
        }
    }
}

File: scaffold/src/main/java/com/vgaw/scaffold/view/TxtTxtLayout.kt
package com.vgaw.scaffold.view;

import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.view.Gravity
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import com.vgaw.scaffold.R
import com.vgaw.scaffold.util.phone.DensityUtil

class TxtTxtLayout : LinearLayout {
    private lateinit var mTxtTxtUp: TextView
    private lateinit var mTxtTxtBelow: TextView

    constructor(context: Context): super(context) {
        init(null)
    }

    constructor(context: Context, attrs: AttributeSet?): super(context, attrs) {
        init(attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int): super(context, attrs, defStyleAttr) {
        init(attrs)
    }

    fun setUpText(up: String) {
        mTxtTxtUp.setText(up)
    }

    private fun init(attrs: AttributeSet?) {
        orientation = VERTICAL
        gravity = Gravity.CENTER

        val view = View.inflate(context, R.layout.txt_txt_layout, this)
        mTxtTxtUp = view.findViewById(R.id.txt_txt_up);
        mTxtTxtBelow = view.findViewById(R.id.txt_txt_below);

        if (attrs != null) {
            val array = context.obtainStyledAttributes(attrs, R.styleable.TxtTxtLayout)
            val txtUp = array.getString(R.styleable.TxtTxtLayout_txtUp)
            val txtBelow = array.getString(R.styleable.TxtTxtLayout_txtBelow)
            val txtUpSize = array.getDimensionPixelSize(R.styleable.TxtTxtLayout_txtUpSize, 0)
            val txtBelowSize = array.getDimensionPixelSize(R.styleable.TxtTxtLayout_txtBelowSize, 0)
            val txtUpColor = array.getColor(R.styleable.TxtTxtLayout_txtUpColor, Color.BLACK)
            val txtBelowColor = array.getColor(R.styleable.TxtTxtLayout_txtBelowColor, Color.BLACK)
            val padding = array.getDimensionPixelSize(R.styleable.TxtTxtLayout_txtPadding, 0)

            array.recycle()

            mTxtTxtUp.text = txtUp
            mTxtTxtUp.textSize = DensityUtil.px2sp(context, txtUpSize.toFloat()).toFloat()
            mTxtTxtUp.setTextColor(txtUpColor)

            mTxtTxtBelow.text = txtBelow
            mTxtTxtBelow.textSize = DensityUtil.px2sp(context, txtBelowSize.toFloat()).toFloat()
            mTxtTxtBelow.setTextColor(txtBelowColor)
            val layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
            layoutParams.topMargin = padding
            mTxtTxtBelow.layoutParams = layoutParams
        }
    }
}
